automata/fa/dfa.py

```diff
@@ -598,6 +598,79 @@ def isfinite(self):
         except nx.exception.NetworkXNoCycle:
             return True
 
+    @classmethod
+    def contains_subsequence(cls, subsequence, input_symbols):
+        """
+        Creates a DFA which accepts all words which contain a specific subsequence of symbols
+        """
+        transitions = {0: {symbol: 0 for symbol in input_symbols}}
+
+        for prev_state, symbol in enumerate(subsequence):
+            next_state = prev_state + 1
+            transitions[next_state] = {symbol: next_state for symbol in input_symbols}
+            transitions[prev_state][symbol] = next_state
+
+        return cls(
+            states=set(transitions.keys()),
+            input_symbols=input_symbols,
+            transitions=transitions,
+            initial_state=0,
+            final_states={len(subsequence)},
+        )
+
+    @classmethod
+    def of_length(cls, min_length, max_length, input_symbols):
+        """
+        Creates a DFA which accepts all words whose length is between `min_length` and `max_length`, inclusive.
+        To allow infinitely long words the value `float('inf')` can be passed in for `max_length`.
+        """
+        transitions = {}
+        length_range = range(min_length) if max_length == float('inf') else range(max_length+1)
```

Comment:
I think it's better to use `None` as a sentinel value, just because using a float means passing in two different types into this function. @caleb531 what do you think?

Reply:
On that note, I think we should make input_symbols the first parameter in these construction methods so we can make easier use of default parameters.

Reply:
Updated the ordering so now `input_symbols` is the first parameter